notebooks/iatraoui.ipynb

```diff
@@ -192,8 +223,33 @@
     "                      0 where no polygon were found, and integers representing classes in order of occurence in the GeoDataFrame\n",
     "    \"\"\"\n",
     "\n",
-    "    ...  # TODO\n",
-    "    raise NotImplementedError\n"
+    "    mapping = {\n",
+    "        'WATER' : 1,\n",
+    "        'FOREST' : 2,\n",
+    "        'FARM' : 3\n",
+    "    }\n",
+    "\n",
+    "    _, x_shape, y_shape = data_array.shape\n",
+    "\n",
+    "    shapes = pd.DataFrame(training_classes)\n",
+    "    shapes = shapes[['geometry', 'class']]\n",
+    "    shapes['class'] = shapes['class'].map(mapping)\n",
+    "    shapes = [tuple(row) for row in shapes.to_numpy()]\n",
+    "    print(shapes[0])\n",
+    "\n",
+    "    out_shape = (x_shape, y_shape)\n",
+    "\n",
+    "    transform = data_array.spatial_ref.GeoTransform\n",
+    "    transform = [float(x) for x in transform.split()]\n",
```

Comment:
Here you can use a generator expression rather than a list comprehension. Indeed, this will be consumed once (in the next line, when unpacking)